pkgs/development/r-modules/default.nix

```diff
@@ -939,6 +938,28 @@ let
   ];
 
   otherOverrides = old: new: {
+    # arrow 14.0.0.2 on CRAN is lagging behind current libarrow release number
+    # https://github.com/apache/arrow/issues/39698
+    arrow = let
+      R-arrow-version = lib.versions.pad 3 (lib.strings.getVersion pkgs.rPackages.arrow);
+      arrow-cpp = pkgs.arrow-cpp.overrideAttrs (finalAttrs: previousAttrs: {
+        version = R-arrow-version;
+        src = fetchurl {
+          url = "mirror://apache/arrow/arrow-${finalAttrs.version}/apache-arrow-${finalAttrs.version}.tar.gz";
+          hash = "sha256-TrDaUOwHG68V/BY8tIBYkx4AbxyGLI3vDhgP0H1TECE=";
+        };
+    });
+    in old.arrow.overrideAttrs (attrs: {
+      postPatch = ''
+        patchShebangs configure
+      '';
+      nativeBuildInputs = [
+        pkgs.pkg-config
+        pkgs.cmake
+        arrow-cpp
+      ] ++ lib.optionals stdenv.isDarwin [ pkgs.intltool ];
```

Comment:
```suggestion
     buildInputs = attrs.buildInputs ++ [ arrow-cpp ];
```

Reply:
PS: it's important to do `attrs.buildInputs ++` when you overrideAttrs, because otherwise you erase any existing buildinput that the package may have. In this case: R and gettext.